benchmarks/decoders/benchmark_decoders_library.py

```diff
@@ -479,13 +480,50 @@ def get_metadata(video_file_path: str) -> VideoStreamMetadata:
     return VideoDecoder(video_file_path).metadata
 
 
+class BatchParameters(NamedTuple):
```

Comment:
Is there a specific reason we're using `NamedTuple` and not a dataclass?

Reply:
I thought it would be lighterweight than dataclass, but I am not too sure. Do you have a preference?

Reply:
I prefer dataclasses. There are some instances where you need to use a namedtuple, but in general, I consider dataclasses to have supplanted namedtuples.